### PR TITLE
[INLONG-4263][Manager] Support HBase sink resource creation

### PR DESCRIPTION
**Pull request by woofyzhao**
Feature #4263

### CODE REVIEW

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sink/hbase/HbaseTableInfo.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+package org.apache.inlong.manager.common.pojo.sink.hbase;
+
+import lombok.Data;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Iceberg table info
+ */
```
Comment (gong) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sink/hbase/HbaseTableInfo.java:27]: change `iceberg` to `hbase`

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/hbase/HbaseApiUtils.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Utils for Iceberg Catalog
+ */
```
Comment (gong) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/hbase/HbaseApiUtils.java:42]: describe error

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/sink/HbaseSinkInfo.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.inlong.sort.protocol.FieldInfo;
+
+import javax.annotation.Nullable;
+
+/**
+ * hbase sink resource info
+ */
+public class HbaseSinkInfo extends SinkInfo {
+
```
Comment (gong) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/sink/HbaseSinkInfo.java:31]: SinkInfo add `@JsonSubTypes` for hbase

Comment (woofyzhao) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/sink/HbaseSinkInfo.java:31]: OK

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/SinkInfoUtils.java [hunk 4]
```diff
@@ -233,4 +243,19 @@ public static void checkPartitionField(List<? extends SinkFieldBase> fieldList,
         }
     }
 
+    private static HbaseSinkInfo createHbaseSinkInfo(HbaseSinkResponse sinkResponse, List<FieldInfo> sinkFields) {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/SinkInfoUtils.java:246]: Add Java doc please.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sink/SinkFieldBase.java [hunk 5]
```diff
@@ -62,4 +62,7 @@ public class SinkFieldBase {
     @ApiModelProperty("Field order")
     private Short rankNum;
 
+    @ApiModelProperty("Column family for columnar storage e.g.hbase")
+    private String columnFamily;
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sink/SinkFieldBase.java:66]: Do we really need to add every specified field to the table when using a different type of sink?

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sink/SinkFieldBase.java (hunk 5 above)
Comment (woofyzhao) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/sink/SinkFieldBase.java:66]: This happens to iceberg table specified fields and i just followed the pattern.

**Review by healchow — APPROVED** [review]

**Review by lucaspeng12138 — APPROVED** [review]

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/sink/SinkInfo.java [hunk 6]
```diff
@@ -41,7 +41,8 @@
         @Type(value = HiveSinkInfo.class, name = Constants.SINK_TYPE_HIVE),
         @Type(value = KafkaSinkInfo.class, name = Constants.SINK_TYPE_KAFKA),
         @Type(value = HiveSinkInfo.class, name = Constants.SINK_TYPE_HIVE),
-        @Type(value = IcebergSinkInfo.class, name = Constants.SINK_TYPE_ICEBERG)}
+        @Type(value = IcebergSinkInfo.class, name = Constants.SINK_TYPE_ICEBERG),
+        @Type(value = IcebergSinkInfo.class, name = Constants.SINK_TYPE_HBASE)}
```
[the remark this answers is not in the document]
Comment (woofyzhao) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/sink/SinkInfo.java]: Already updated. please refresh.

**Pull request merged**